tools/analyzer_plugin/lib/src/component_usage.dart

```diff
@@ -153,7 +153,12 @@ bool hasChildComponent(ArgumentList arguments) {
 }
 
 /// Attempt to find and return the closest expression that encloses the [node]
-/// and is an independent Flutter `Widget`.  Return `null` if nothing found.
+/// and is an independent Flutter `Widget`.
+///
+/// This utility is intended to be used when [node] is a part of a component invocation
+/// (i.e. is a component [InvocationExpression] or part of the props [CascadeExpression]).
+///
+/// Returns `null` if nothing is found or if [node] is a child.
```

Comment:
Updated the doc comment to reflect how the utility is used in existing diagnostics.